components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -658,6 +662,8 @@ void AdsServiceImpl::OnInitialize(
 
   is_initialized_ = true;
 
+  SetAdsServiceForNotificationHandler();
```

Comment:
@yachtcaptain23 This was only initialized for native notifications, this should be guarded against dev or canary builds